src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java

```diff
@@ -97,7 +95,8 @@ private SurfaceTransformerContext getSurfaceTransformerContextFromService(Interf
         service,
         getApiConfig(),
         modelTypeTable,
-        new NodeJSSurfaceNamer(getApiConfig().getPackageName()));
+        new NodeJSSurfaceNamer(getApiConfig().getPackageName()),
+        false);
```

Comment:
Ok I'm about to dump some work on you here...

It's hard to know what this "false" means in code like this. In general I'd like to avoid booleans as function parameters.

At a higher level, I think we might have a developing case for something like a FeatureConfig set of classes. Each language would have to provide a subclass (similar to SurfaceNamer). We should definitely put your new setting into it (resourceNameTypesEnabled), and any other globally-applicable features in the future would go in there too.